pages/jsx.md

```diff
@@ -8,11 +8,463 @@ hideInToc: true
+
+---
+hideInToc: true
+transition: fade
+---
+
+[Syntax and Structure of JSX]{.text-gradient.text-4xl}
+
+The syntax of JSX elements looks very much alike with that of HTML elements.
+
+For example, if you want to create a `h1` and a `div` using JSX, they will look the same as with normal HTML:
+
+```jsx
+<h1>Hello AltSchoolers 👋, This is a JSX heading and I love it!</h1>
+
+<div>Alright, important question. How about centering a div in JSX? 🤔</div>
+```
+
+<div flex flex-col gap-3 mt-3 v-click>
+
+```jsx
+// With JSX
+const myElement = <h1>You know, I Love JSX!</h1>
+
+const root = ReactDOM.createRoot(document.getElementById('root'))
+root.render(myElement)
+```
+
+```js
+// Without JSX
+const myElement = React.createElement('h1', {}, 'He said he does not use JSX!')
+
+const root = ReactDOM.createRoot(document.getElementById('root'))
+root.render(myElement)
+```
+
+</div>
+
+---
+hideInToc: true
+transition: slide-down
+---
+
+[The]{.text-4xl} [Rules]{.text-gradient.text-4xl} [of]{.text-4xl} [JSX]{.text-gradient.text-4xl}
+
+There are few rules that guides writing JSX. They include:
+
+1. Return a single root element: This is simply saying you should always wrap a group of JSX elements with a single root.
+
+```jsx {*|2|6|*}
+// the `<div></div>` is the parent (root element)
+<div>
+  <h1>I am the first Element in the group!</h1>
+  <p>Well I am the second, but I have details more than you do!</p>
+  <img src="imageUrl" alt="I am the bigger picture guys.">
+</div>
+```
+
+If you don’t want to add an extra `<div>` to your markup, you can use `<>` and `</>` instead:
+
+```jsx {*|2|6|*}
+// the `<></>` is the parent (root element) called Fragment
+<>
+  <h1>Our parent Element don't even have a public name</h1>
+  <p>Well, everyone knows the name is <b>Fragment</b>!</p>
+  <img src="imageUrl" alt="<p> my paddy is always right">
+</>
+```
+
+---
+hideInToc: true
+level: 2
+transition: view-transition
+---
+
+2. Close all the tags: In JSX, every tag should be explicitly closed. This means that all tags, including self-closing tags like `<img>`, `<link>`, and `<meta>`, should be explicitly closed with the forward-slash (/): `<img />`, `<link />`, and `<meta />`. JSX follows XML rules, and therefore HTML elements must be properly closed. JSX will throw an error if the HTML is not properly closed.
+
+```jsx {hide|3|4|5|6|*}
+// Closing tags properly
+<>
+  <h1>Making sure all tags are explicitly closed</h1>
+  <p>I have got some good news!</p>
+  <img src="imageUrl" alt="Thank you for closing me properly in JSX" />
+  <input type="text" />
+</>
+```
+
+<div v-click>
+3. Use camelCase for most of the things!: JSX converts to JavaScript, and attributes written in JSX become keys of JavaScript objects. As we know, JavaScript has limitations on variable names, hence using camelCase is recommended. For example, their names can’t contain dashes or be reserved words like `class`. Since `class` is a reserved word, in React you write `className` instead.
```

Comment:
Move no 3 to a separate slide @Stan015. It is cutting out in the preview link.

Reply:
Done ✅